docs/c-grader/index.md

```diff
@@ -92,77 +92,77 @@ To compile a C or C++ file, you may use the methods `self.compile_file()` or `se
-self.test_compile_file('square.c', 'square', add_warning_result_msg=False) # Creates a test result that includes the warning as the output
+self.compile_file("square.c", "square", add_warning_result_msg=False)      # Does not show any warnings
+self.test_compile_file("square.c", "square", add_warning_result_msg=False) # Creates a test result that includes the warning as the output
 ```
 
 You may also include additional compilation flags accepted by `gcc` (or `g++`) with the `flags` argument, which can be invoked with a single string for flags, or with an array of strings:
 
 ```python
-self.compile_file('square.c', 'square', flags='-Wall -O3') # single string
-self.compile_file('square.c', 'square', flags=['-Wall', '-O3']) # array
+self.compile_file("square.c", "square", flags="-Wall -O3") # single string
+self.compile_file("square.c", "square", flags=["-Wall", "-O3"]) # array
 ```
 
 For flags based on `pkg-config`, use `pkg_config_flags` with the library or libraries that should be queried.
 
 ```python
-self.compile_file('square.c', 'square', pkg_config_flags='check ncurses') # single string
-self.compile_file('square.c', 'square', pkg_config_flags=['check', 'ncurses']) # array
+self.compile_file("square.c", "square", pkg_config_flags="check ncurses") # single string
+self.compile_file("square.c", "square", pkg_config_flags=["check", "ncurses"]) # array
 ```
 
 It is also possible to test programs where the student only submits part of an application. To compile the C/C++ file submitted by the user with some functions (including, for example, a `main` function) implemented by the instructor, you can save a `main.c` (or `main.cpp`) file inside the `tests` folder and run:
 
 ```python
-self.compile_file('square.c', 'square', add_c_file='/grade/tests/main.c')
+self.compile_file("square.c", "square", add_c_file="/grade/tests/main.c")
 ```
 
 The instruction above will compile the student-provided C/C++ file with the instructor-provided C/C++ file into the same executable. If the student implements any function that is also implemented by the instructor's code, including `main` if provided, it will be ignored, and the instructor-provided functions will take precedence. This is achieved by compiling the code with the `-Wl,--allow-multiple-definition` flag of `gcc`.
 
 It is also possible to compile multiple student files and multiple question-provided files into a single executable, by providing lists of files:
 
 ```python
-self.compile_file(['student_file1.c', 'student_file2.c'], 'executable',
-                  add_c_file=['/grade/tests/question_file1.c',
-                              '/grade/tests/question_file2.c'],
-                  flags=['-I/grade/tests', '-I/grade/student'])
+self.compile_file(["student_file1.c", "student_file2.c"], "executable",
+                  add_c_file=["/grade/tests/question_file1.c",
+                              "/grade/tests/question_file2.c"],
+                  flags=["-I/grade/tests", "-I/grade/student"])
```

Comment:
It'd be super nice to get Prettier formatting these snippets in Markdown files with black, though I don't know how easy that would be. In principle, Prettier lets us add plugins to support other languages. Maybe it wouldn't be too hard to do that?

Only bringing this up because black would format this differently:

```suggestion
self.compile_file(
    ["student_file1.c", "student_file2.c"],
    "executable",
    add_c_file=["/grade/tests/question_file1.c", "/grade/tests/question_file2.c"],
    flags=["-I/grade/tests", "-I/grade/student"],
)
```

(https://black.vercel.app/ is great for quickly black-ifying some code)

Reply:
I think the issue is that you have to get Prettier to run black on Python code fences in markdown. I'm not sure if there's a plugin or anything for that; some preliminary googling hasn't revealed much. It would definitely be nice, I can take a closer look later.